src/variables.ts

```diff
@@ -12,9 +12,20 @@ type getInputObjectTypeRec<
       Rest,
       Introspection,
       (InputField extends { name: any; type: any }
-        ? InputField extends { defaultValue: undefined | null; type: { kind: 'NON_NULL' } }
-          ? { [Name in InputField['name']]: unwrapType<InputField['type'], Introspection> }
-          : { [Name in InputField['name']]?: unwrapType<InputField['type'], Introspection> | null }
+        ? InputField extends { type: { kind: 'NON_NULL' } }
+          ? InputField extends { defaultValue: infer DefaultValue }
+            ? DefaultValue extends undefined | null
+              ? { [Name in InputField['name']]: unwrapType<InputField['type'], Introspection> }
+              : {
+                  [Name in InputField['name']]?: unwrapType<
+                    InputField['type'],
+                    Introspection
+                  > | null;
+                }
+            : { [Name in InputField['name']]: unwrapType<InputField['type'], Introspection> }
+          : {
+              [Name in InputField['name']]?: unwrapType<InputField['type'], Introspection> | null;
+            }
```

Comment:
```suggestion
        ? InputField extends { defaultValue?: undefined | null; type: { kind: 'NON_NULL' } }
          ? { [Name in InputField['name']]: unwrapType<InputField['type'], Introspection> }
          : { [Name in InputField['name']]?: unwrapType<InputField['type'], Introspection> | null }
```

This is a bit of a wordy change and complexity increase in the types.
What about this change? It seems to basically be able to do the same thing and be equivalent.